src/promptflow-devkit/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -78,3 +86,16 @@ def post(self):
         )
         # Todo : remove output_path when exit executor which is registered in pfs
         return result
+
+
+@api.route("/infer_signature")
+class FlowInferSignature(Resource):
+    @api.response(code=200, description="Flow infer signature", model=dict_field)
+    @api.doc(description="Flow infer signature")
+    @api.expect(flow_infer_signature_parser)
+    def post(self):
+        args = flow_infer_signature_parser.parse_args()
+        flow_path = decrypt_flow_path(args.source)
+        flow = load_flow(source=flow_path)
+        infer_signature = get_client_from_request().flows._infer_signature(entry=flow)
```

Comment:
it seems that we can't handle dag flow now. Maybe we can ask extension to keep their previous logic to handle dag flow first?

Reply:
offline sync with extension